shell/src/main/java/alluxio/cli/fs/command/CpCommand.java

```diff
@@ -61,6 +67,66 @@
           .hasArg(false)
           .desc("copy files in subdirectories recursively")
           .build();
+  private static final int NUM_THREADS = Runtime.getRuntime().availableProcessors() * 2;
+  private static final String COPY_PROGRESS_DONE = "#";
+
+  private ThreadPoolExecutor mCopyExecutor;
+  private Phaser mCopyDoneSignal;
+  private BlockingQueue<String> mCopyProgress;
+  private Thread mCopyProgressDisplayThread;
+
+  /**
+   * Initializes necessary resources for async copy.
+   */
+  private void initAsyncCopy() {
+    mCopyExecutor = new ThreadPoolExecutor(NUM_THREADS, NUM_THREADS,
```

Comment:
make the number of threads configurable? perhaps pass in as `alluxio fs copyFromLocal -D<property_name>=<num_threads>`

Reply:
done